.claude/component_testing_patterns.md

```diff
@@ -0,0 +1,427 @@
+# Component Testing Patterns
+
+This document provides comprehensive guidance for testing Dagster components (classes that extend `dagster.Component`). These patterns are based on analysis of existing component tests throughout the Dagster codebase.
+
+## Core Testing Infrastructure
```

Comment:
how much of this content is in docs md files? I think we could potentially use `@` linking to avoid duplicate content that might desync

Reply:
Yeah this is also a lot of tokens